File: tests/bindings/types_invalid.ts
import { Client, MultipartPart } from 'tlsurl'

new Client({ platform: 'not-a-platform' })
new Client({ tls: { minVersion: '9.9' } })
new Client().get('https://example.org', { timeotMs: 1 })
new Client().post('https://example.org', { bodyFile: 123 })
const missing: MultipartPart = { name: 'missing-source' }
const conflict: MultipartPart = { name: 'conflict', data: 'text', file: 'file.bin' }
void missing
void conflict

File: tests/bindings/types_invalid.py
"""Each line below must be rejected by the installed Python type declarations."""
from tlsurl import Client

Client(platform="not-a-platform")
Client(tls={"min_version": "9.9"})
Client().get("https://example.org", timeot_ms=1)
Client().post("https://example.org", body_file=123)
Client().post("https://example.org", multipart=[{"name": "missing-source"}])
